src/Config.h

```diff
@@ -0,0 +1,1488 @@
+                    return false;
+                }
+
+                setBackflushFillTime(value);
+            }
+
+            if (doc["backflush"].containsKey("flush_time")) {
+                const double value = doc["backflush"]["flush_time"].as<double>();
+
+                if (!validateParameterRange("backflush.flush_time", value, BACKFLUSH_FLUSH_TIME_MIN, BACKFLUSH_FLUSH_TIME_MAX)) {
+                    return false;
+                }
+
+                setBackflushFlushTime(value);
+            }
+
+            // Standby parameters
+            if (doc["standby"].containsKey("enabled")) {
+                if (!doc["standby"]["enabled"].is<bool>()) {
+                    return false;
+                }
+
+                setStandbyModeOn(doc["standby"]["enabled"].as<bool>());
+            }
+
+            if (doc["standby"].containsKey("time")) {
+                const double value = doc["standby"]["time"].as<double>();
+
+                if (!validateParameterRange("standby.time", value, STANDBY_MODE_TIME_MIN, STANDBY_MODE_TIME_MAX)) {
+                    return false;
+                }
+
+                setStandbyModeTime(value);
+            }
+
+            // Feature parameters
+            if (doc["features"].containsKey("brew_control")) {
+                if (!doc["features"]["brew_control"].is<bool>()) {
+                    return false;
+                }
+
+                setFeatureBrewControl(doc["features"]["brew_control"].as<bool>());
+            }
+
+            if (doc["display"].containsKey("fullscreen_brew_timer")) {
+                if (!doc["display"]["fullscreen_brew_timer"].is<bool>()) {
+                    return false;
+                }
+
+                setFeatureFullscreenBrewTimer(doc["features"]["fullscreen_brew_timer"].as<bool>());
+            }
+
+            if (doc["display"].containsKey("fullscreen_manual_flush_timer")) {
+                if (!doc["display"]["fullscreen_manual_flush_timer"].is<bool>()) {
+                    return false;
+                }
+
+                setFeatureFullscreenManualFlushTimer(doc["display"]["fullscreen_manual_flush_timer"].as<bool>());
+            }
+
+            if (doc["display"].containsKey("post_brew_timer_duration")) {
+                const double value = doc["display"]["post_brew_timer_duration"].as<double>();
+
+                if (!validateParameterRange("display.post_brew_timer_duration", value, POST_BREW_TIMER_DURATION_MIN, POST_BREW_TIMER_DURATION_MAX)) {
+                    return false;
+                }
+
+                setPostBrewTimerDuration(value);
+            }
+
+            if (doc["display"].containsKey("heating_logo")) {
+                if (!doc["display"]["heating_logo"].is<bool>()) {
+                    return false;
+                }
+
+                setFeatureHeatingLogo(doc["display"]["heating_logo"].as<bool>());
+            }
```

Comment:
I am trying to understand what this function for example does. There are two options:

* `doc["display"]["heating_logo"]` **is** a boolean: The first condition returns false, function continues. We read the value of `doc["display"]["heating_logo"]` and assign it to `doc["display"]["heating_logo"]`. This is a no-op...?
* `doc["display"]["heating_logo"]` **is not** a boolean: first condition kicks in, function aborts, value stays a non-boolean JSON value.

What am I missing?

Reply:
oh, `doc` and `_doc`, now I see. Okay, still this might be a bit of a cumbersome validation to do for every parameter - and if the input is not correct, the value will not be set.